python/pyspark/cloudpickle.py

```diff
@@ -852,11 +853,11 @@ def _rebuild_tornado_coroutine(func):
 
 # Shorthands for legacy support
 
-def dump(obj, file, protocol=2):
+def dump(obj, file, protocol=protocol):
```

Comment:
@inpefess, I was actually meaning upgrade cloudpickle to upper version see https://github.com/apache/spark/pull/21350. Our copy is matched to 0.4.4 for now.

The main reason we stayed in 0.4.x was that the default pickle protocol was changed as of 0.5.x.
Since we're targeting to upgrade it here, It should be good to match cloudpickle copy to higher version.

Reply:
@HyukjinKwon upgrading cloudpickle copy is a different task, although related to this one. I suggest upgrading protocol here and creating a separate issue for the cloudpickle upgrade.

Reply:
Ok, I'll copy the newest cloudpickle version then.